### PR TITLE
Merges changes to the GettyImages shortcode from WordPress.com.

### LINKED ISSUE #8569: Shortcodes: Sync Getty shortcode updates from WPCOM
Getty has updated their embed markup syntax. The changes necessary to support the new syntax have been made on WordPress.com, and should be similarly synced to Jetpack.

Commits:

- r159940-wpcom
- r160012-wpcom
- r162170-wpcom
- r168553-wpcom

Context: p5j4vm-1y7-p2

### PR DESCRIPTION
**Pull request by zinigor**
This is a merge of changes from WordPress.com related to the new format of markup returned by GettyImages oembed endpoints. D9498-wpcom is the mirror change.

Fixes #8569 

#### Testing instructions:
Make sure Getty Images shortcode works as expected in both ways:
* You should be able to insert an embed code from Getty Images;
* You should be able to use a shortcode like this: `[getty src="82278805" width="462" height="370"]`

Both ways should result in an embedded image.

#### Proposed changelog entry for your changes:
* Updated the Getty Images shortcode to match changes made on their end.

### CODE REVIEW

File: modules/shortcodes/getty.php [hunk 1]
```diff
@@ -47,39 +82,73 @@ function jetpack_getty_enable_embeds( $site = 'wpcom' ) {
...
 
-	$regexp = '!<iframe\s+src=[\'"](https?:)?//embed\.gettyimages\.com/embed(/|/?\?assets=)(\d+(,\d+)*)[^\'"]*?[\'"]((?:\s+\w+=[\'"][^\'"]*[\'"])*)((?:[\s\w]*))></iframe>!i';
+	$regexp = '!<iframe\s+src=[\'"](https?:)?//embed\.gettyimages\.com/embed(/|/?\?assets=)([a-z0-9_-]+(,[a-z0-9_-]+)*)[^\'"]*?[\'"]((?:\s+\w+=[\'"][^\'"]*[\'"])*)((?:[\s\w]*))></iframe>!i';
 	$regexp_ent = str_replace( '&amp;#0*58;', '&amp;#0*58;|&#0*58;', htmlspecialchars( $regexp, ENT_NOQUOTES ) );
 
-	foreach ( array( 'regexp', 'regexp_ent' ) as $reg ) {
+	/**
+	 * Context: http://wp.me/p5j4vm-1y7
+	 *
+	 * @hide-in-jetpack
```
Comment (jeherve) [inline modules/shortcodes/getty.php]: Looks like this made it in when it should not have?

Comment (zinigor) [reply modules/shortcodes/getty.php]: Thank you, removed!

File: modules/shortcodes/getty.php [hunk 2]
```diff
@@ -27,8 +27,8 @@ function jetpack_getty_enable_embeds( $site = 'wpcom' ) {
 		: 'wordpress.com';
 
 	// Support their oEmbed Endpoint
-	wp_oembed_add_provider( '#https?://www\.gettyimages\.com/detail/.*#i', "https://embed.gettyimages.com/oembed/?caller=$caller", true );
```
Comment (aduth) [inline modules/shortcodes/getty.php]: This makes the `$caller` variable unused? Should probably be removed.

**Comment by zinigor** [comment]
Thanks for taking a look, Andrew! I have fixed caller handling.

**Pull request merged**